server/tests/backends/sql_translator/test_sql_postgres_pypika_translator_steps.py

```diff
@@ -0,0 +1,172 @@
+import json
+import logging
+import time
+from typing import List, Optional
+
+import docker
+import pandas as pd
+import psycopg2
+import pytest
+from docker.models.images import Image
+from psycopg2 import OperationalError
+from sqlalchemy import create_engine
+
+from tests.utils import assert_dataframes_equals, get_spec_from_json_fixture, retrieve_case
+from weaverbird.backends.pypika_translator.dialects import SQLDialect
+from weaverbird.backends.pypika_translator.translate import translate_pipeline
+from weaverbird.pipeline import PipelineWithVariables
+
+image = {'name': 'postgres_weaverbird_test', 'image': 'postgres', 'version': '14.1-bullseye'}
+docker_client = docker.from_env()
+exec_type = 'postgres_pypika'
+
+test_cases = retrieve_case('sql_translator', exec_type)
+
+
+@pytest.fixture(scope='module', autouse=True)
+def db_container():
+    images: List[Image] = docker_client.images.list()
+    found = False
+    for i in images:
+        try:
+            if i.tags[0] == f'{image["image"]}:{image["version"]}':
+                found = True
+        except IndexError:
+            pass
+    if not found:
+        logging.getLogger(__name__).info(
+            f'Download docker image {image["image"]}:{image["version"]}'
+        )
+        docker_client.images.pull(f'{image["image"]}:{image["version"]}')
+
+    logging.getLogger(__name__).info(f'Start docker image {image["image"]}:{image["version"]}')
+    docker_container = docker_client.containers.run(
+        image=f'{image["image"]}:{image["version"]}',
+        name=f'{image["name"]}',
+        auto_remove=True,
+        detach=True,
+        environment={
+            'POSTGRES_DB': 'pg_db',
+            'POSTGRES_USER': 'ubuntu',
+            'POSTGRES_PASSWORD': 'ilovetoucan',
+        },
+        ports={'5432': '5432'},
+    )
+    ready = False
+    while not ready:
+        time.sleep(1)
+        try:
+            if docker_container.status == 'created' and get_connection():
+                ready = True
+        except OperationalError:
+            pass
+    yield docker_container
```

Comment:
Could we please have a timeout here ? or set the `timeout-minutes` parameter in the CI config file ? We've struggled with infinite loops in the CI before :grimacing: